//ESTA ES UNA FUNCION QUE RECIBE UNA FUNCION, Y RETORNA UNA FUNCION,
//LA FUNCION QUE RETORNA RECIBE REQ, RES Y NEXT, y EJECUTA LA FUNCION QUE RECIBE CON REQ Y RES
//DE ESTA ULTIMA FUNCION SE CATCHEA EL ERROR, Y SE MANDA A EXPRESS CON NEXT
const catchedAsync = (fn) => {
  return (req, res, next) => {
    fn(req, res).catch((err) => next(err));
  };
};

//ESTA FUNCION ES UNA FUNCION DE ORDEN SUPERIOR EN JAVASCRIPT PORQUE RECIBE UNA FUNCION Y/O RETORNA UNA FUNCION

module.exports = catchedAsync;
